build/xUnit.props

```diff
@@ -0,0 +1,31 @@
+<Project TreatAsLocalProperty="TestReportFileName;DotnetErrorCode">
+
+  <PropertyGroup>
+    <TestReportFileName>$(MSBuildProjectName)</TestReportFileName>
+    <TestReportFileName Condition="'$(TestReportFileName)' == ''">test</TestReportFileName>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$([System.Text.RegularExpressions.Regex]::IsMatch(`$(TargetFramework)`, `^netcoreapp`))' == 'True'">
+    <!-- `dotnet xunit` requires a very explicit runtime version to be set: -->
+    <RuntimeFrameworkVersion>2.0.3</RuntimeFrameworkVersion>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.7.2" PrivateAssets="All" />
+    <PackageReference Include="xunit" Version="2.3.1" PrivateAssets="All" />
+    <PackageReference Include="xunit.runner.visualstudio" Version="2.3.1" PrivateAssets="All" />
+    <DotNetCliToolReference Include="dotnet-xunit" Version="2.3.1" PrivateAssets="All" />
+  </ItemGroup>
+
+  <Target Name="Test">
+    <!-- See if dotnet CLI is available: -->
+    <Exec ContinueOnError="True" Command="dotnet --info">
+      <Output TaskParameter="ExitCode" PropertyName="DotnetErrorCode"/>
+    </Exec>
+    <Warning Condition="'$(DotnetErrorCode)' != '0'" Text="Error in executing dotnet." />
+
+    <Exec Command="dotnet xunit -nobuild -configuration $(Configuration) -noshadow -appveyor -html &quot;$(OutputDirectory)$(TestReportFileName).html&quot; -xml &quot;$(OutputDirectory)$(TestReportFileName).xml&quot;"
```

Comment:
Not sure why we need this instead of just using `xunit.runners.msbuild` and passing in the test assembly...

Reply:
Because that runner targets only `net452`. That is, won't be usable on Linux boxes without resorting to Mono. I would love to have a build script that runs on Linux, too.

Btw. I'm looking at replacing `dotnet xunit` with `xunit.runner.console`. The former is going away with xUnit 4.

Reply:
hm. I wonder how many request to run this from Linux, honestly... But fine then!

Reply:
Well, we're supporting .NET Core so why not do it thoroughly. :)

And it's not just Linux. It's also relevant for those who (accidentally) try to build using `dotnet msbuild` instead of `msbuild`. Both should work equally well, IMO.